packages/node/src/models/free-balance.ts

```diff
@@ -52,14 +55,122 @@ export type TokenIndexedCoinTransferMap = {
   [tokenAddress: string]: CoinTransferMap;
 };
 
+// todo(xuanji): replace with Set
 export type ActiveAppsMap = { [appInstanceIdentityHash: string]: true };
 
-export type FreeBalanceState = {
+export class FreeBalanceClass {
+  private constructor(
+    private readonly activeAppsMap: ActiveAppsMap,
+    private readonly balancesIndexedByToken: {
```

Comment:
Why not use `tokenIndexedBalances`?

Reply:
I will probably do this in a followup PR, but this class was meant to replicate `FreeBalanceState` since that was the closest thing to a good data structure being passed around

Reply:
Ok